flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/LeaderElectionService.java

```diff
@@ -67,4 +67,20 @@
 	 * @return true if the associated {@link LeaderContender} is the leader, otherwise false
 	 */
 	boolean hasLeadership();
+
+	/**
+	 * [EXPERIMENTAL] Returns <code>true</code> if the {@link LeaderContender} is leader. A call
+	 * to this method might block.
+	 *
+	 * <p>This forces a synchronous check at the respective state backend. It is possible
+	 * that is does not reflect the current state at the {@link LeaderContender}, which is notified
+	 * asynchronously. Therefore it is possible that {@link #hasLeadership()} and {@link
+	 * #syncHasLeadership()} have different return values.
+	 *
+	 * @TODO @tillrohrmann Is it OK to collapse this with {@link #hasLeadership()}?
+	 *
+	 * @return true if the associated {@link LeaderContender} is the leader, otherwise false
+	 */
+	boolean syncHasLeadership();
```

Comment:
I do not know, whether this method adds much value. The problem is that reading the leader path data and comparing the leader session ID do not happen atomically. Thus, the moment you've just read the leader path, it can happen that the node's data and, thus, the leader changed. IMO, this is the same problem as with `leaderLatch.hasLeadership`.